### PR TITLE
Adding a process lifecycle with permission restrictions

### PR DESCRIPTION
**Pull request by dewmini**

### CODE REVIEW

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/extensions/aspects/ProcessLifeCycle.java [hunk 1]
```diff
@@ -0,0 +1,883 @@
+/*
```
Comment (warunalakshitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/extensions/aspects/ProcessLifeCycle.java]: Remove unwanted spaces between codelines

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+        try {
+            user = context.getResource().getAuthorUserName();
+            String[] roles = CurrentSession.getUserRealm().getUserStoreManager().getRoleNames();
+            path = context.getResourcePath().toString();
+            userRegistry = (UserRegistry) context.getSystemRegistry();
+
+            setPublisherPermission(currentState, targetState);
+            for (String role : roles) {
+                if (role.equals("admin") || role.equals("Internal/publisher") || role
```
Comment (rmsamitha) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java]: It is better to replace these roles with constants

Comment (dewmini) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/extensions/aspects/ProcessLifeCycle.java (hunk 1 above)
Comment (dewmini) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/extensions/aspects/ProcessLifeCycle.java]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+        } else if (currentState.equals("In-Review") && targetState.equals("Development")) {
+            assignPermission("Internal/publisher", "+get,+add,+delete,+authorize");
+
+        } else if (currentState.equals("In-Review") && targetState.equals("Published")) {
+            assignPermission("Internal/publisher", "+get,+add,-delete,+authorize");
+
+        } else if (currentState.equals("Published") && targetState.equals("Development")) {
+            assignPermission("Internal/publisher", "+get,+add,+delete,+authorize");
+
```
Comment (dewmini) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

File: modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        return updated;
+    }
+
+    /**
+     * This method is used to identify the permissions to be assigned and denied from the permission string and add or
+     * remove the permission.
+     *
+     * @param role
+     * @param permissionString
```
Comment (dewmini) [inline modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/executors/LCExecutor.java]: Fixed in https://github.com/wso2/product-pc/commit/e931bb704e9065cf57f08f61619a45a4beecb5c7

**Pull request merged**